build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
++            mUrl = view.findViewById(R.id.url);
++
++            mMoreButton = view.findViewById(R.id.more);
++        }
++
++        protected void updateScriptInfo(ScriptInfo item) {
++            mSwitch.setOnCheckedChangeListener(null);
++            mSwitch.setChecked(item.Enabled);
++            mSwitch.setOnCheckedChangeListener(mOnOffListener);
++
++            mTitle.setText(item.Name);
++            mDescription.setText(item.Description);
++            mVersion.setText(item.Version);
++            mFile.setText(item.Key);
++            mUrl.setText(item.UrlSource);
++
++            if (item.UrlSource == null || item.UrlSource.isEmpty()) {
++                mUrl.setVisibility(View.GONE);
++            }
++        }
++
++        void setMenuButtonDelegate(@NonNull ListMenuButtonDelegate delegate) {
++            mMoreButton.setVisibility(View.VISIBLE);
++            mMoreButton.setDelegate(delegate);
++            // Set item row end padding 0 when MenuButton is visible.
++            ViewCompat.setPaddingRelative(itemView, ViewCompat.getPaddingStart(itemView),
++                    itemView.getPaddingTop(), 0, itemView.getPaddingBottom());
++        }
++
++        void setItemListener(@NonNull ItemClickListener listener) {
++            mOnOffListener = (buttonView, isChecked) -> listener.onScriptOnOff(isChecked);
++            mSwitch.setOnCheckedChangeListener(mOnOffListener);
++            itemView.setOnClickListener(view -> listener.onScriptClicked());
++        }
++    }
++
++    private class ScriptDragStateDelegate implements DragStateDelegate {
++        private AccessibilityManager mA11yManager;
++        private AccessibilityStateChangeListener mA11yListener;
++        private boolean mA11yEnabled;
++
++        public ScriptDragStateDelegate() {
++            mA11yManager =
++                    (AccessibilityManager) mContext.getSystemService(Context.ACCESSIBILITY_SERVICE);
++            mA11yEnabled = mA11yManager.isEnabled();
++            mA11yListener = enabled -> {
++                mA11yEnabled = enabled;
++                notifyDataSetChanged();
++            };
++            mA11yManager.addAccessibilityStateChangeListener(mA11yListener);
++        }
++
++        @Override
++        public boolean getDragEnabled() {
++            return !mA11yEnabled;
++        }
++
++        @Override
++        public boolean getDragActive() {
++            return getDragEnabled();
++        }
++
++        @Override
++        public void setA11yStateForTesting(boolean a11yEnabled) {
++        }
++    }
++
++    ScriptListBaseAdapter(Context context) {
++        super(context);
++        setDragStateDelegate(new ScriptDragStateDelegate());
++    }
++
++    void showDragIndicatorInRow(ScriptInfoRowViewHolder holder) {
++        // Quit if it's not applicable.
++        if (getItemCount() <= 1 || !mDragStateDelegate.getDragEnabled()) return;
++
++        // assert mItemTouchHelper != null;
```

Comment:
Can we remove this commented block here? What would it do, if uncommented?

Reply:
> Can we remove this commented block here? 

yes, the idea was to give the possibility to reorder the list with drag & drop, but it's completely useless, I remove it

Reply:
Why useless? It sounds nice

Reply:
removed in https://github.com/bromite/bromite/pull/857/commits/8c164138391082350f20493a0ca5c4926887372c

>  Actually, the order of scripts might be important for functionality.

the insertion is asynchronous and out of the possibility of inserting a precise order without complicating life

Reply:
So the scripts start all at once? This is important for documentation

Reply:
no, one at a time, only the order is not fixed

Reply:
Ok, this is all information that should go in the README and also in the wiki page.

Reply:
seriously? so technical? the code is there to see ...
ahh I understand, it will take me a lifetime ...

Reply:
The scripts show an order on the UI, from top to bottom. Hence the user will be **certain** that they are executed in such order.

It has to be documented.